libultraship/Makefile.switch

```diff
@@ -0,0 +1,191 @@
+#---------------------------------------------------------------------------------
+.SUFFIXES:
+#---------------------------------------------------------------------------------
+
+ifeq ($(strip $(DEVKITPRO)),)
+$(error "Please set DEVKITPRO in your environment. export DEVKITPRO=<path to>/devkitpro")
+endif
+
+TOPDIR ?= $(CURDIR)
+include $(DEVKITPRO)/libnx/switch_rules
+
+#---------------------------------------------------------------------------------
+# TARGET is the name of the output
+# BUILD is the directory where object files & intermediate files will be placed
+# SOURCES is a list of directories containing source code
+# DATA is a list of directories containing data files
+# INCLUDES is a list of directories containing header files
+# ROMFS is the directory containing data to be added to RomFS, relative to the Makefile (Optional)
+#
+# NO_ICON: if set to anything, do not use icon.
+# NO_NACP: if set to anything, no .nacp file is generated.
+# APP_TITLE is the name of the app stored in the .nacp file (Optional)
+# APP_AUTHOR is the author of the app stored in the .nacp file (Optional)
+# APP_VERSION is the version of the app stored in the .nacp file (Optional)
+# APP_TITLEID is the titleID of the app stored in the .nacp file (Optional)
+# ICON is the filename of the icon (.jpg), relative to the project folder.
+#   If not set, it attempts to use one of the following (in this order):
+#     - <Project name>.jpg
+#     - icon.jpg
+#     - <libnx folder>/default_icon.jpg
+#
+# CONFIG_JSON is the filename of the NPDM config file (.json), relative to the project folder.
+#   If not set, it attempts to use one of the following (in this order):
+#     - <Project name>.json
+#     - config.json
+#   If a JSON file is provided or autodetected, an ExeFS PFS0 (.nsp) is built instead
+#   of a homebrew executable (.nro). This is intended to be used for sysmodules.
+#   NACP building is skipped as well.
+#---------------------------------------------------------------------------------
+TARGET		:=	ultraship
+BUILD		:=	build
+SOURCES		:=
+DATA		:=
+INCLUDES	:=	\
+		../ZAPDTR/ZAPDUtils \
+		../StormLib/src \
+		libultraship/Lib/Fast3D/U64 \
+		libultraship/Lib/ImGui \
+		libultraship/Lib/spdlog \
+		libultraship/Lib/spdlog/include \
+		libultraship
+
+#-------------------------------------------------------------------------------
+# source files
+#-------------------------------------------------------------------------------
+SOURCEFILES_C :=    \
+		libultraship/mixer.c \
+		libultraship/Lib/stb/stb_impl.c \
+
+SOURCEFILES_CPP :=	\
+	$(shell find libultraship/Factories -name "*.cpp") \
```

Comment:
You can add these directories to the `SOURCES` variable to avoid extra shell commands in here.